File: java-takeaway-challenge/app/src/main/java/OrderSystem/CartFormatter.java
package OrderSystem;

public class CartFormatter {
  private Cart cart = new Cart();

  public CartFormatter(Cart cart) {
    this.cart = cart;
  }

  public void renderCart() {
    for (CartItem item : cart.getItems()) {
      System.out.printf("%d x %s: £%.2f\n", item.getQuantity(), item.getName(), item.getPrice());
    }
  }
}

File: golden-square-solo-project/lib/cart_formatter.rb
class CartFormatter
  def initialize(cart)
    @cart = cart
  end

  def format
    formatted_output = @cart.get_items.map.with_index{|item, i| "#{i+1}. #{item.name}: £#{item.price}"}.join("\n")
    formatted_output += "\n"
    formatted_output += "Total: £#{'%.2f' % @cart.get_items.sum{|item| item.price}}"
    return formatted_output
  end
end